[INSTALL] intel-openmpi on nautilus

Please add intel-openmpi stack on nautilus.  Using intelmpi-intel on AMD Epyc Rome infiniband machines produces a deadlock with all2all type communications (JCSDA-internal/oops#2191).  The problem has also been reported with ESMF.  Using openmpi instead of intelmpi fixes the problem with ESMF.

**Which software in the stack would you like installed?**
Please provide the name of the software you would like installed on the HPCs.  Ensure the software exists in the list of supported software.  If new software needs to be added to the stack, use the [add_package_request.md](./add_package_request.md) template.

**What is the version/tag of the software?**
Please provide the version/tag of the software to be installed.

**What compilation options would you like set?**
Give a detailed list of compilation options, compiler and MPI flavors and versions, etc.

**Which machines would you like to have the software installed?**
See currently managed HPC machines.

**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

@areinecke Quick update. I am hitting a compilation error with Intel + OpenMPI for parallel-netcdf (which is required for parallelio, esmf, ...). I'll need some time to debug this:
```
7 errors found in build log:
     722    mv -f .deps/ncmpiFile.Tpo .deps/ncmpiFile.Plo
     723    mv -f .deps/ncmpiVlenType.Tpo .deps/ncmpiVlenType.Plo
     724    mv -f .deps/ncmpiVarAtt.Tpo .deps/ncmpiVarAtt.Plo
     725    libtool: compile:  /opt/scyld/openmpi/4.1.4/intel/bin/mpic++ -DHAVE_CONFIG_H -I. -I../../../src/include -I../../../src/include -fPIC -MT ncmpiVar.lo -MD -MP -MF .deps/ncmpiVar.Tpo -c ncmpiVar.cpp -o
            ncmpiVar.o >/dev/null 2>&1
     726    mv -f .deps/ncmpiVar.Tpo .deps/ncmpiVar.Plo
     727    /bin/sh ../../../libtool  --tag=CXX   --mode=link /opt/scyld/openmpi/4.1.4/intel/bin/mpic++  -fPIC   -o libcxx.la  ncmpi_notyet.lo ncmpiCheck.lo ncmpiType.lo ncmpiAtt.lo ncmpiEnumType.lo ncmpiGroupAt
            t.lo ncmpiGroup.lo ncmpiByte.lo ncmpiChar.lo ncmpiCompoundType.lo ncmpiDouble.lo ncmpiFloat.lo ncmpiInt.lo ncmpiInt64.lo ncmpiOpaqueType.lo ncmpiShort.lo ncmpiUbyte.lo ncmpiUint.lo ncmpiUint64.lo ncm
            piUshort.lo ncmpiException.lo ncmpiDim.lo ncmpiFile.lo ncmpiVarAtt.lo ncmpiVar.lo ncmpiVlenType.lo
  >> 728    /usr/bin/grep: /opt/scyld/pbs/lib/libpbs.la: No such file or directory
  >> 729    /usr/bin/sed: can't read /opt/scyld/pbs/lib/libpbs.la: No such file or directory
     730    libtool:   error: '/opt/scyld/pbs/lib/libpbs.la' is not a valid libtool archive
  >> 731    make[4]: *** [Makefile:566: libcxx.la] Error 1
     732    make[4]: Leaving directory '/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/cache/build_stage/heinzell/spack-stage-parallel-netcdf-1.12.2-trtldc5y2ygbiabr7c4loidzymk4dlmv/spack-src/src/binding/
            cxx'
  >> 733    make[3]: *** [Makefile:520: all] Error 2
     734    make[3]: Leaving directory '/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/cache/build_stage/heinzell/spack-stage-parallel-netcdf-1.12.2-trtldc5y2ygbiabr7c4loidzymk4dlmv/spack-src/src/binding/
            cxx'
  >> 735    make[2]: *** [Makefile:474: all-recursive] Error 1
     736    make[2]: Leaving directory '/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/cache/build_stage/heinzell/spack-stage-parallel-netcdf-1.12.2-trtldc5y2ygbiabr7c4loidzymk4dlmv/spack-src/src/binding'
  >> 737    make[1]: *** [Makefile:476: all-recursive] Error 1
     738    make[1]: Leaving directory '/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/cache/build_stage/heinzell/spack-stage-parallel-netcdf-1.12.2-trtldc5y2ygbiabr7c4loidzymk4dlmv/spack-src/src'
  >> 739    make: *** [Makefile:533: all-recursive] Error 1
```

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,9 +1,11 @@
 [submodule "spack"]
   path = spack
-  #url = https://github.com/spack/spack
-  #branch = develop
-  url = https://github.com/jcsda/spack
-  branch = jcsda_emc_spack_stack
+  ##url = https://github.com/spack/spack
+  ##branch = develop
+  #url = https://github.com/jcsda/spack
+  #branch = jcsda_emc_spack_stack
+  url = https://github.com/climbfuji/spack
+  branch = feature/jedi_neptune_env_update_20230628
 [submodule "doc/CMakeModules"]
   path = doc/CMakeModules
   url = https://github.com/noaa-emc/cmakemodules

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,11 +1,9 @@
 [submodule "spack"]
   path = spack
-  ##url = https://github.com/spack/spack
-  ##branch = develop
-  #url = https://github.com/jcsda/spack
-  #branch = jcsda_emc_spack_stack
-  url = https://github.com/climbfuji/spack
-  branch = feature/jedi_neptune_env_update_20230628
+  #url = https://github.com/spack/spack
+  #branch = develop
+  url = https://github.com/jcsda/spack
+  branch = jcsda_emc_spack_stack
 [submodule "doc/CMakeModules"]
   path = doc/CMakeModules
   url = https://github.com/noaa-emc/cmakemodules

--- a/configs/sites/nautilus/compilers.yaml
+++ b/configs/sites/nautilus/compilers.yaml
@@ -33,7 +33,8 @@ compilers:
       prepend_path:
         PATH: '/opt/rh/gcc-toolset-11/root/usr/bin'
         CPATH: '/opt/rh/gcc-toolset-11/root/usr/include'
-        LD_LIBRARY_PATH: '/p/app/compilers/intel/oneapi/compiler/2022.0.2/linux/compiler/lib/intel64_lin:/opt/rh/gcc-toolset-11/root/usr/lib64:/opt/rh/gcc-toolset-11/root/usr/lib'
-      set:
-        I_MPI_ROOT: '/p/app/compilers/intel/oneapi/mpi/2021.5.1'
+        LD_LIBRARY_PATH: '/opt/scyld/slurm/lib64:/opt/scyld/slurm/lib64/slurm:/p/app/compilers/intel/oneapi/compiler/2022.0.2/linux/compiler/lib/intel64_lin:/opt/rh/gcc-toolset-11/root/usr/lib64:/opt/rh/gcc-toolset-11/root/usr/lib'
+      # For intel-oneapi-mpi only
+      #set:
+      #  I_MPI_ROOT: '/p/app/compilers/intel/oneapi/mpi/2021.5.1'
     extra_rpaths: []

--- a/configs/sites/nautilus/packages.yaml
+++ b/configs/sites/nautilus/packages.yaml
@@ -3,6 +3,8 @@ packages:
     compiler:: [intel@2021.5.0, aocc@4.0.0]
     providers:
       mpi:: [openmpi@4.1.5, openmpi@4.1.4]
+      blas:: [intel-oneapi-mkl]
+      lapack:: [intel-oneapi-mkl]
 
 ### MPI, Python, MKL
   mpi:
@@ -21,6 +23,12 @@ packages:
       - penguin/openmpi/4.1.5rc2/intel
     - spec: openmpi@4.1.4%aocc@4.0.0~cuda~cxx~cxx_exceptions~java~memchecker+pmi~static~wrapper-rpath fabrics=ucx schedulers=slurm
       prefix: /p/app/penguin/openmpi/4.1.4/aoc
+  intel-oneapi-mkl:
+    externals:
+    - spec: intel-oneapi-mkl@2022.0.2
+      prefix: /p/app/compilers/intel/oneapi/mkl/2022.0.2
+      modules:
+      - intel-oneapi-mkl@2022.0.2
 
 ### Modifications of common packages
   # Version 2.0.8 doesn't compile on Nautilus

--- a/configs/sites/nautilus/packages.yaml
+++ b/configs/sites/nautilus/packages.yaml
@@ -7,6 +7,7 @@ packages:
       mpi:: [openmpi@4.1.5rc2]
       #mpi:: [openmpi@4.1.4]
       blas:: [intel-oneapi-mkl]
+      fftw-api:: [intel-oneapi-mkl]
       lapack:: [intel-oneapi-mkl]
   ectrans:
     variants:: +mkl

--- a/configs/sites/nautilus/packages.yaml
+++ b/configs/sites/nautilus/packages.yaml
@@ -2,19 +2,23 @@ packages:
   all:
     compiler:: [intel@2021.5.0, aocc@4.0.0]
     providers:
-      mpi:: [intel-oneapi-mpi@2021.5.1, openmpi@4.1.4]
+      mpi:: [openmpi@4.1.5, openmpi@4.1.4]
 
 ### MPI, Python, MKL
   mpi:
     buildable: False
-  intel-oneapi-mpi:
-    externals:
-    - spec: intel-oneapi-mpi@2021.5.1%intel@2021.5.0
-      modules:
-      - intel/mpi/2021.5.1
-      prefix: /p/app/compilers/intel/oneapi
+  #intel-oneapi-mpi:
+  #  externals:
+  #  - spec: intel-oneapi-mpi@2021.5.1%intel@2021.5.0
+  #    modules:
+  #    - intel/mpi/2021.5.1
+  #    prefix: /p/app/compilers/intel/oneapi
   openmpi:
     externals:
+    - spec: openmpi@4.1.5rc2%intel@2021.5.0~cuda~cxx~cxx_exceptions~java~memchecker+pmi~static~wrapper-rpath fabrics=ucx schedulers=slurm
+      prefix: /p/app/penguin/openmpi/4.1.5rc2/intel
+      modules:
+      - penguin/openmpi/4.1.5rc2/intel
     - spec: openmpi@4.1.4%aocc@4.0.0~cuda~cxx~cxx_exceptions~java~memchecker+pmi~static~wrapper-rpath fabrics=ucx schedulers=slurm
       prefix: /p/app/penguin/openmpi/4.1.4/aoc
 

--- a/configs/sites/nautilus/packages.yaml
+++ b/configs/sites/nautilus/packages.yaml
@@ -2,9 +2,16 @@ packages:
   all:
     compiler:: [intel@2021.5.0, aocc@4.0.0]
     providers:
-      mpi:: [openmpi@4.1.5, openmpi@4.1.4]
+      # For now need to enable one or the other;
+      # see https://github.com/JCSDA/spack-stack/issues/659
+      mpi:: [openmpi@4.1.5rc2]
+      #mpi:: [openmpi@4.1.4]
       blas:: [intel-oneapi-mkl]
       lapack:: [intel-oneapi-mkl]
+  ectrans:
+    variants:: +mkl
+  gsibec:
+    variants:: +mkl
 
 ### MPI, Python, MKL
   mpi:
@@ -21,12 +28,16 @@ packages:
       prefix: /p/app/penguin/openmpi/4.1.5rc2/intel
       modules:
       - penguin/openmpi/4.1.5rc2/intel
+      - slurm
     - spec: openmpi@4.1.4%aocc@4.0.0~cuda~cxx~cxx_exceptions~java~memchecker+pmi~static~wrapper-rpath fabrics=ucx schedulers=slurm
       prefix: /p/app/penguin/openmpi/4.1.4/aoc
+      modules:
+      - penguin/openmpi/4.1.4/aocc
+      - slurm
   intel-oneapi-mkl:
     externals:
     - spec: intel-oneapi-mkl@2022.0.2
-      prefix: /p/app/compilers/intel/oneapi/mkl/2022.0.2
+      prefix: /p/app/compilers/intel/oneapi
       modules:
       - intel-oneapi-mkl@2022.0.2
 

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -30,7 +30,7 @@ Ready-to-use spack-stack 1.4.0 installations are available on the following, ful
 +------------------------------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------------+
 | NAVY HPCMP Narwhal GNU^**                                  | Dom Heinzeller / ???          | ``/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-gcc-10.3.0``                        |
 +------------------------------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------------+
-| NAVY HPCMP Nautilus Intel^*                                | Dom Heinzeller / ???          | ``/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-intel-2021.5.0-hdf5-1.14.0``        |
+| NAVY HPCMP Nautilus Intel^*                                | Dom Heinzeller / ???          | ``/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-intel-2021.5.0-openmpi-4.1.5``      |
 +------------------------------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------------+
 | NAVY HPCMP Nautilus AMD clang/flang                        | Dom Heinzeller / ???          | **currently not supported**                                                                                  |
 +------------------------------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------------+
@@ -255,10 +255,9 @@ With Intel, the following is required for building new spack environments and fo
 .. code-block:: console
 
    module purge
-
    module load slurm
    module load intel/compiler/2022.0.2
-   module load intel/mpi/2021.5.1
+   module load penguin/openmpi/4.1.5rc2/intel
 
    module use /p/app/projects/NEPTUNE/spack-stack/modulefiles
    module load ecflow/5.8.4
@@ -268,9 +267,9 @@ For ``spack-stack-1.4.0`` with Intel, load the following modules after loading t
 
 .. code-block:: console
 
-   module use /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-intel-2021.5.0-hdf5-1.14.0/install/modulefiles/Core
+   module use /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-intel-2021.5.0-openmpi-4.1.5/install/modulefiles/Core
    module load stack-intel/2021.5.0
-   module load stack-intel-oneapi-mpi/2021.5.1
+   module load stack-openmpi/4.1.5rc2
    module load stack-python/3.10.8
 
 With AMD clang/flang (aocc), the following is required for building new spack environments and for using spack to build and run software.

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -188,6 +188,7 @@ With Intel, the following is required for building new spack environments and fo
 
 .. code-block:: console
 
+   umask 0022
    module unload PrgEnv-cray
    module load PrgEnv-intel/8.3.2
    module unload intel
@@ -216,6 +217,7 @@ With GNU, the following is required for building new spack environments and for 
 
 .. code-block:: console
 
+   umask 0022
    module unload PrgEnv-cray
    module load PrgEnv-gnu/8.3.2
    module unload gcc
@@ -254,7 +256,9 @@ With Intel, the following is required for building new spack environments and fo
 
 .. code-block:: console
 
+   umask 0022
    module purge
+
    module load slurm
    module load intel/compiler/2022.0.2
    module load penguin/openmpi/4.1.5rc2/intel
@@ -276,6 +280,7 @@ With AMD clang/flang (aocc), the following is required for building new spack en
 
 .. code-block:: console
 
+   umask 0022
    module purge
 
    module load slurm

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -271,7 +271,7 @@ For ``spack-stack-1.4.0`` with Intel, load the following modules after loading t
 
 .. code-block:: console
 
-   module use /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.4.0/envs/unified-env-intel-2021.5.0-openmpi-4.1.5/install/modulefiles/Core
+   module use /p/app/projects/NEPTUNE/spack-stack/spack-stack-dev-20230628/envs/unified-env/install/modulefiles/Core
    module load stack-intel/2021.5.0
    module load stack-openmpi/4.1.5rc2
    module load stack-python/3.10.8

